controllers/src/main/java/com/redhat/ipaas/controllers/IntegrationController.java

```diff
@@ -114,7 +117,9 @@ private void checkIntegration(Integration integration) {
         Optional<Integration.Status> current = integration.getCurrentStatus();
         if (desired.isPresent() && !current.equals(desired)) {
             WorkflowHandler workflowHandler = handlers.get(desired.get());
-            if (workflowHandler != null) {
+
+            String scheduledKey = "" + integration.getDesiredStatus() + ":" + integration.getId().get();
```

Comment:
The integration id can be (and was) null. See e.g. https://github.com/rhuss/ipaas-rest/blob/c5b788abb7b2fc53514717bb3c61eae1192a7ffb/rest/src/main/java/com/redhat/ipaas/rest/v1/handler/integration/IntegrationHandler.java#L120-L120 for a usage where the id is null.